scIB/metrics.py

```diff
@@ -1667,22 +1667,28 @@ def kBET(adata, batch_key, label_key, embed='X_pca', type_ = None,
 
 # determine root cell for trajectory conservation metric
 def get_root(adata_pre, adata_post, ct_key, dpt_dim=3):
+    
     n_components, adata_post.obs['neighborhood'] = connected_components(csgraph=adata_post.uns['neighbors']['connectivities'], directed=False, return_labels=True)
     
     start_clust = adata_pre.obs.groupby([ct_key]).mean()['dpt_pseudotime'].idxmin()
-    min_dpt = np.flatnonzero(adata_pre.obs[ct_key] == start_clust)
-    max_neigh = np.flatnonzero(adata_post.obs['neighborhood']== adata_post.obs['neighborhood'].value_counts().argmax())
+    min_dpt = adata_pre.obs[adata_pre.obs[ct_key] == start_clust].index
+    #print(min_dpt)
+    max_neigh = adata_post.obs[adata_post.obs['neighborhood']== adata_post.obs['neighborhood'].value_counts().idxmax()].index
```

Comment:
this looks like it still does the same as above... why change?

Reply:
the old one returned the integer index, the new one uses the obs_name. I think the integer index changes when we subset the post integration object to a different size as the pre_integration object so it would change. Another option would probably be to subset adata_pre to the largest connected component as well.

Reply:
Why did we ever work with the integer index? Good to fix this.

Reply:
I think this was due to changing `obs_names` between pre and post integration for some methods

Reply:
I see. So as we have now fixed this in the metrics.py script, it doesn't matter anymore. Good stuff.